### PR TITLE
`ProfileCantVoteCard` component

### PR DESCRIPTION
**Pull request by NoahSaso**
This displays when the user cannot vote on the proposal they're viewing.

They may or may not be a member currently, so different content will show depending on whether or not they currently have staked tokens.

If they have no staked tokens right now:
image

If they DO have staked tokens right now:
Screen Shot 2022-08-24 at 8 04 59 PM

And the tooltip helps make this clearer as well:
Screen Shot 2022-08-24 at 8 05 13 PM

### CODE REVIEW

**Comment by NoahSaso** [comment]
I also moved membership content shared by some cards, relevant to the cw20-staked-balance-voting module, into its own component so it will be easy to move it into the voting module adapter.

File: packages/ui/components/profile/ProfileCantVoteCard.tsx [hunk 1]
```diff
@@ -1,5 +1,71 @@
...
+
+        <div className="flex flex-row justify-between items-center mt-3 secondary-text">
+          <p>{t('title.vote')}</p>
+          <p className="text-text-interactive-error">{t('info.none')}</p>
+        </div>
+      </div>
+
+      <Cw20StakedBalanceVotingProfileMembership
+        className="p-6 border-t border-t-border-primary"
+        junoswapHref="https://junoswap.com"
```
Comment (elsehow) [inline packages/ui/components/profile/ProfileCantVoteCard.tsx:58]: Maybe out of scope for this PR, but: do we have a good way of getting the pools programatically?

File: packages/ui/components/Button/ButtonLink.tsx [hunk 2]
```diff
@@ -8,21 +8,15 @@ import {
   getButtonifiedClassNames,
 } from './Buttonifier'
 
-export type ButtonLinkProps = ComponentPropsWithoutRef<'a'> &
-  ButtonifierProps & {
-    remote?: boolean
-  }
+export type ButtonLinkProps = ComponentPropsWithoutRef<'a'> & ButtonifierProps
```
[the remark this answers is not in the document]
Comment (NoahSaso) [reply packages/ui/components/Button/ButtonLink.tsx:11]: This is TypeScript! `ComponentPropsWithoutRef` is a generic type that takes a string, which is the name of an HTML tag. `ref` is a React thing where you can set variables to be references to DOM elements so you can run vanilla JS functions on them directly. Check [this page](https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring) out for an example on that type.

`ButtonifierProps` is another interface (in the Buttonifier.tsx class) that takes various styling props, and we want to include all those in the `ButtonLinkProps` so they will work. `ButtonLink` and `Button` both wrap the `ButtonifiedChildren` component which uses those props.

File: packages/i18n/locales/en/translation.json [hunk 3]
```diff
@@ -333,6 +334,7 @@
     "member": {
       "memberOf": "Member of {{daoName}}"
     },
+    "memberCantVote": "You were not a member of {{daoName}} when this proposal was created, so you cannot vote. You will still be able to vote on future proposals, as long as you have ${{tokenSymbol}} tokens staked.",
```
[the remark this answers is not in the document]
Comment (NoahSaso) [reply packages/i18n/locales/en/translation.json:337]: Love it. Also agreed about the tooltips. I don't think we should have a separate root key for "profile" like we do now.

File: packages/ui/components/profile/ProfileCantVoteCard.tsx (hunk 1 above)
Comment (NoahSaso) [reply packages/ui/components/profile/ProfileCantVoteCard.tsx:58]: Yeah we should look into that! I'll add a TODO.

**Pull request merged**